src/LiveComponent/src/ComponentWithFormTrait.php

```diff
@@ -164,9 +160,10 @@ private function submitForm(bool $validateAll = true): void
     private function extractFormValues(FormView $formView): array
     {
         $values = [];
+
         foreach ($formView->children as $child) {
             $name = $child->vars['name'];
-            if (\count($child->children) > 0) {
+            if (!($child->vars['expanded'] ?? false) && \count($child->children) > 0) {
```

Comment:
this deserves a small `// comment` above to help understand, as it's complex. I'm actually not entirely sure what this should be doing - I think it is (correctly) fixing ChoiceType with expanded => true, but it's very complex.

Reply:
Actually, I stole your logic over in #264: https://github.com/symfony/ux/pull/264/files#diff-9d445f9954f64a41536bb9470d1abcea4986100b295a038022b55ba192a85cf7R187

It DOES fix a critical bug - nice work! Please keep this here... and also, I didn't include a test for that functionality in my PR, but I believe you have one here.

Reply:
Right, there is already a test in my PR for that :+1: